src/abstract-list-field/abstract-list-field.component.ts

```diff
@@ -75,12 +76,20 @@ export abstract class AbstractListFieldComponent extends AbstractFieldComponent
   /**
    * Returns path of the property of an element at index.
    */
-  getValuePath(index: number, property: string): Array<any> {
-    return this.path.concat(index, property);
+  getValuePath(index: number, property?: string): Array<any> {
+    let valuePathString = this.getElementPathString(index, property);
+    if (!this.paths[valuePathString]) {
+      this.paths[valuePathString] = property ? this.path.concat(index, property) : this.path.concat(index);
+    }
+    return this.paths[valuePathString];
   }
 
-  getElementPathString(index: number): string {
-    return `${this.pathString}${this.pathUtilService.separator}${index}`;
+  getElementPathString(index: number,  property?: string): string {
```

Comment:
Please don't change this, and in the [Line#80](https://github.com/inveniosoftware-contrib/ng2-json-editor/pull/196/files#diff-0831221ed39b574c418645ef0262336eR80) you can do use it like below:

```typescript
`${this.getElementPathString(index)}${this.pathUtilService.separator}${property || ''}`
```

Because this place is the only case where we pass property to the this function.